### PR TITLE
Add unit tests for toXContent methods in ReplicationResponse

### PR DESCRIPTION
**Pull request by tlrx**
This commit adds unit tests for the `toXContent()` methods of the inner classes  `ReplicationResponse.ShardInfo` and `ReplicationResponse.ShardInfo.Failure`.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/DocWriteResponse.java [hunk 1]
```diff
@@ -242,7 +242,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (forcedRefresh) {
             builder.field("forced_refresh", forcedRefresh);
         }
-        shardInfo.toXContent(builder, params);
+        builder.field("_shards", shardInfo);
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/action/DocWriteResponse.java]: why did you move the field name `_shards` out?

File: core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java [hunk 2]
```diff
@@ -113,6 +115,156 @@ public void testFailureEqualsAndHashcode() {
...
+            + "}", builder.string());
+        }
+    }
+
+    public void testRandomShardInfoToXContent() throws IOException {
+        final ReplicationResponse.ShardInfo shardInfo = randomShardInfo();
+
+        // Build the expected JSON from the random ShardInfo
+        StringBuilder expectedJson = new StringBuilder();
+        expectedJson.append("{");
```
Comment (javanna) [inline core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java]: why do we build json by hand here?

Comment (tlrx) [inline core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java]: I find it easier to read... would you prefer using a jsonBuilder()?

File: core/src/main/java/org/elasticsearch/action/DocWriteResponse.java (hunk 1 above)
Comment (tlrx) [inline core/src/main/java/org/elasticsearch/action/DocWriteResponse.java]: I did it so that ShardInfo.toXContent() starts with a startObject() and not a field name, but it is not a requirement. It's just easier to test since there's no need to wrap it into an anonymous object.

Comment (javanna) [inline core/src/main/java/org/elasticsearch/action/DocWriteResponse.java]: ah right because otherwise it is not a valid json alone. I don't know, I think that if we touch all these little things as part of this parsing effort it becomes risky. I did that myself in my PR and I am thinking of reverting it. Maybe we should just accept that toXContent may print something that's not a valid json alone, that's it. It kinda makes sense that shard info starts its own object with its own name. I find it odd when toXContent simply starts fields without an ancestor.  What do you think?

File: core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java (hunk 2 above)
Comment (javanna) [inline core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java]: I see, I was hoping we could also test other formats, we test only json here? using a builder we could test also other formats?

Comment (javanna) [inline core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java]: or maybe we have other tests in this class that already use other formats, in which case it's ok as is

File: core/src/main/java/org/elasticsearch/action/DocWriteResponse.java (hunk 1 above)
Comment (tlrx) [inline core/src/main/java/org/elasticsearch/action/DocWriteResponse.java]: I like when toXContent() starts it's own anonymous object because it makes it easy to include the object anywhere: array, field, as root object etc. But I understand your concern and in this case I agree it makes sense for ShardInfo to starts its own _shards object. So I reverted the change.

File: core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java (hunk 2 above)
Comment (tlrx) [inline core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java]: I changed that so that the xcontent type is randomized.

**Comment by tlrx** [comment]
@javanna I udated the PR. Hope this is better now.

File: core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java [hunk 3]
```diff
@@ -113,6 +116,157 @@ public void testFailureEqualsAndHashcode() {
...
+    }
+
+    public void testRandomShardInfoToXContent() throws IOException {
+        final ReplicationResponse.ShardInfo shardInfo = randomShardInfo();
+
+        final XContent xContent = randomFrom(XContentType.values()).xContent();
+        try (XContentBuilder builder = XContentBuilder.builder(xContent)) {
+            builder.startObject();
+            shardInfo.toXContent(builder, ToXContent.EMPTY_PARAMS);
+            builder.endObject();
```
Comment (javanna) [inline core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java]: I am adding a util method for this in XContentHelper, maybe worth replacing this later (should not block this PR)

**Review by javanna — APPROVED** [review]
LGTM

**Comment by tlrx** [comment]
Thanks @javanna
